orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -221,6 +223,184 @@ func (s *OrderbookUsecaseTestSuite) TestProcessPool() {
+						{
+							OrderId: 1,
+						},
+					}, 1, nil
+
+				}
+
+				withGetMetadataByChainDenom(tokensusecase)
+
+				// Set is best effort to true
+				orderbookrepository.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{}, false
+				}
+			},
+			expectedError:        nil,
+			expectedIsBestEffort: true,
+		},
+		{
+			name: "successful retrieval of active orders",
+			setupContext: func() context.Context {
+				return context.Background()
+			},
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, grpcclient *mocks.OrderbookGRPCClientMock, poolsUsecase *mocks.PoolsUsecaseMock, tokensusecase *mocks.TokensUsecaseMock) {
+				withGetAllCanonicalOrderbookPoolIDs(poolsUsecase)
+
+				grpcclient.MockGetActiveOrdersCb = func(ctx context.Context, contractAddress string, ownerAddress string) (orderbookdomain.Orders, uint64, error) {
+					return orderbookdomain.Orders{
+						{
+							TickId:         cltypes.MaxTick,
+							OrderId:        1,
+							OrderDirection: "bid",
+							Owner:          "owner1",
+							Quantity:       "1000",
+							PlacedQuantity: "1500",
+							Etas:           "500",
+							ClaimBounty:    "10",
+							PlacedAt:       "1634764800000",
+						},
+					}, 1, nil
+
+				}
+
+				withGetMetadataByChainDenom(tokensusecase)
+
+				tokensusecase.GetSpotPriceScalingFactorByDenomFunc = func(baseDenom, quoteDenom string) (osmomath.Dec, error) {
+					return osmomath.NewDec(1), nil
+				}
+
+				orderbookrepository.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{
+						TickState: orderbookdomain.TickState{
+							BidValues: orderbookdomain.TickValues{
+								EffectiveTotalAmountSwapped: "100",
+							},
+						},
+						UnrealizedCancels: orderbookdomain.UnrealizedCancels{
+							BidUnrealizedCancels: osmomath.NewInt(100),
+						},
+					}, true
+				}
+			},
+			expectedError:        nil,
+			expectedIsBestEffort: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Create instances of the mocks
+			poolsUsecase := mocks.PoolsUsecaseMock{}
+			orderbookRepository := mocks.OrderbookRepositoryMock{}
+			client := mocks.OrderbookGRPCClientMock{}
+			tokensusecase := mocks.TokensUsecaseMock{}
+
+			// Setup the mocks according to the test case
+			usecase := orderbookusecase.New(&orderbookRepository, &client, &poolsUsecase, &tokensusecase, &log.NoOpLogger{})
+			if tc.setupMocks != nil {
+				tc.setupMocks(usecase, &orderbookRepository, &client, &poolsUsecase, &tokensusecase)
+			}
+
+			ctx := tc.setupContext()
+
+			// Call the method under test
+			// We are not interested in the orders returned, it's tested
+			// in the TestCreateFormattedLimitOrder.
+			_, isBestEffort, err := usecase.GetActiveOrders(ctx, "address")
```

Comment:
Can we also assert the limit order return as opposed to ignoring it?

At the very least we could validate that the number of orders is as expected and IDs match

Reply:
Implemented to assert actual orders with expected ones. :+1: